scipy/io/mmio.py

```diff
@@ -317,8 +373,23 @@ def __init__(self, **kwargs):
         self._init_attrs(**kwargs)
 
     #---------------------------------------------------------------------------
-    def read(self, source):
-        stream, close_it = self._open(source)
+    def read(self, filename):
+        """
+        Reads the contents of a Matrix Market file 'filename' into a matrix.
+    
+        Parameters
+        ----------
+    
+        filename : file
```

Comment:
I think the Python doc is confusing here; in Python 2 there is a class `file`, which is only for files on file systems.  So, I think `file-like` is considerably better to convey something with a `read, write, seek, tell` interface, which is the meaning here. 

So, I suggest something like:

```
source : str or file-like
```

in the docstring.